### PR TITLE
Document config initialization with envs

### PR DESCRIPTION
**Pull request by vk496**
Documentation of feature from https://github.com/ipfs/go-ipfs/pull/8326

### CODE REVIEW

File: docs/how-to/run-ipfs-inside-docker.md [hunk 1]
```diff
@@ -64,6 +64,20 @@ When starting a container running ipfs for the first time with an empty data dir
 ```shell
 docker run -d --name ipfs_host -e IPFS_PROFILE=server -v $ipfs_staging:/export -v $ipfs_data:/data/ipfs -p 4001:4001 -p 4001:4001/udp -p 127.0.0.1:8080:8080 -p 127.0.0.1:5001:5001 ipfs/go-ipfs:latest
 ```
+## Environmet override
```
Comment (johnnymatthews) [inline docs/how-to/run-ipfs-inside-docker.md:67]:
```suggestion
## Environmet override

```

Title's need a single empty line before and after.

File: docs/how-to/run-ipfs-inside-docker.md [hunk 2]
```diff
@@ -64,6 +64,20 @@ When starting a container running ipfs for the first time with an empty data dir
...
+You can override the default configuration during the *first* initialization with variables starting with **IPFCONFIG** and *_* as separators, which will be passed directly to `ipfs config`
```
Comment (johnnymatthews) [inline docs/how-to/run-ipfs-inside-docker.md:68]:
```suggestion
You can override the default configuration during the first initialization with variables starting with `IPFCONFIG` and `_` as separators. These variables are passed directly to `ipfs config`:
```

File: docs/how-to/run-ipfs-inside-docker.md [hunk 3]
```diff
@@ -64,6 +64,20 @@ When starting a container running ipfs for the first time with an empty data dir
...
+
+For example:
+```shell
+docker run --rm -e IPFSCONFIG_Swarm_EnableAutoRelay=true -e IPFSCONFIG_Addresses_Swarm=[\"/ip4/0.0.0.0/tcp/4002\",\"/ip6/::/tcp/4002\"] ipfs/go-ipfs:latest
+```
```
Comment (johnnymatthews) [inline docs/how-to/run-ipfs-inside-docker.md:73]:
````suggestion
```shell
docker run --rm -e IPFSCONFIG_Swarm_EnableAutoRelay=true -e IPFSCONFIG_Addresses_Swarm=[\"/ip4/0.0.0.0/tcp/4002\",\"/ip6/::/tcp/4002\"] ipfs/go-ipfs:latest
```
````

File: docs/how-to/run-ipfs-inside-docker.md [hunk 4]
```diff
@@ -64,6 +64,20 @@ When starting a container running ipfs for the first time with an empty data dir
...
+
+Some important notes:
+ - The names are case sensitive. `IPFSCONFIG_FooBar=true` and `IPFSCONFIG_foobar=true` are not the same
+ - Arrays must use escaped double quotes, like `IPFSCONFIG_FooBar=[\"test\"]`
+ - Syntactically valid config will **not** cause any error, even if it does anything. `IPFSCONFIG_zzz_abcds=true` wil be written to the config
+ - Invalid variables will be ignored, like `IPFSCONFIG_foobar_=true`
+ - Invalid values will faill initialization, like `IPFSCONFIG_foobar=trues`
```
Comment (johnnymatthews) [inline docs/how-to/run-ipfs-inside-docker.md:80]:
```suggestion
Configuration variables must follow these rules:

- The names are case sensitive. `IPFSCONFIG_saturn=true` and `IPFS_CONFIG_Saturn=true` are not the same, so adding both to the configuration file will create two separate variables.
- Arrays must use escaped double quotes:

    ```shell
    IPFSCONFIG_FooBar=[\"test\"]
    ```

- Syntactically valid config will **not** cause any errors.
    - Invalid variables will be ignored. For example, `IPFSCONFIG_pluto_=true` will _not_ create a `pluto_` variable.
    - Invalid values will fail initialization. For example, `IPFSCONFIG_foobar=trues` will cause IPFS to output an error.
```

**Review by johnnymatthews — APPROVED** [review]
Thanks for your help here @vk496. I've made some minor tweaks.

## Blocked

Gonna wait on https://github.com/ipfs/go-ipfs/pull/8326 before this gets merged.

**Comment by vk496** [comment]
Thank you @johnnymatthews 

Just one thing that I was not sure how to deal: indicate minimum required version.

Most Docker users expect environment override as a basic feature. They will see the instructions for using it but also they have ALL previous images without that feature ready to use in the docker hub.

**Comment by johnnymatthews** [comment]
> indicate minimum required version - @vk496 

Minimum version of IPFS? Or Docker? Or both?

**Comment by vk496** [comment]
> Minimum version of IPFS? Or Docker? Or both?

Docker images of IPFS. https://hub.docker.com/r/ipfs/go-ipfs/tags

**Comment by lidel** [comment]
We decided on a different approach, as this one had some gaps.
See https://github.com/ipfs/go-ipfs/pull/8326#issuecomment-912726852

**Pull request closed without merging**